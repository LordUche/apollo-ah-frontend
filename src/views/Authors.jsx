/* eslint-disable react/no-array-index-key */
/* eslint-disable react/require-default-props */
/* eslint-disable react/button-has-type */
import React from 'react';
import PropTypes from 'prop-types';

/**
 * @description - returns all author's or selected author's
 * @param {props} authors - all authors
 * @param {props} begin - start of range of author's to select
 * @param {props} end - end of range of author's to select
 * @param {props} follow - follow feature
 * @return {JSX}
 */
const Authors = ({ authors, begin, end, follow }) => {
  return (
    <div className="container">
      <div className="row">
        {
          authors.length > 0 ?
            authors.slice(begin, end).map((author, index) => {
              const { firstname, lastname } = author.authorsProfile;
              return (
                <div className="col-sm-6 col-lg-4" key={author.id}>
                  <div className="author">
                    <img alt="author" src={author.authorsProfile.image} />
                    <div data-pg-collapsed>
                      <h4>{`${firstname} ${lastname}`}</h4>
                      <p>{author.authorsProfile.bio}</p>
                      <button
                        onClick={follow}
                        className="transition btn"
                        type="button"
                      >
                        Follow
                      </button>
                    </div>
                  </div>
                </div>
              );
            }) : ''
        }
      </div>
    </div>
  </div>
);

Authors.propTypes = {
  authors: PropTypes.array,
  begin: PropTypes.number,
  end: PropTypes.number,
  follow: PropTypes.func,
};

Authors.defaultProps = {
  authors: [],
  begin: 1,
  end: 10,
  follow: f => f,
};

export default Authors;
