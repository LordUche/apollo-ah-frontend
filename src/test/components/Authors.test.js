/* eslint-disable react/jsx-filename-extension */
import React from 'react';
import { shallow } from 'enzyme';
import Authors from '../../views/Category';

const props = [{ 
  begin: 0,
  end: 6,
  authors: [{
    authorsProfile: {
      image: 'https://images.unsplash.com/photo-1478358161113-b0e11994a36b?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjEyMDd9&auto=format&fit=crop&w=800&q=60',
      firstname: 'Andra',
      lastname: 'Collins',
      bio: 'I love writing inspirational articles',
    },
  }],
}];

describe('<Authors Test Suite>', () => {
  describe('<Authors>', () => {
    it('It should render succesfully', () => {
      const wrapper = shallow(<Authors {...props} />);
      expect(wrapper).toBeDefined();
      expect(wrapper.length).toBe(1);
      expect(wrapper).toMatchSnapshot();
    });
  });
});
