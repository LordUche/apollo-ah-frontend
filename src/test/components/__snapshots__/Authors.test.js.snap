// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Authors Test Suite> <Authors> It should render succesfully 1`] = `ShallowWrapper {}`;
